dot/sync/chain_sync.go

```diff
@@ -932,6 +936,44 @@ func (cs *chainSync) validateJustification(bd *types.BlockData) error {
 	return nil
 }
 
+func (cs *chainSync) getHighestBlock() (int64, error) {
+	cs.RLock()
+	ps := cs.peerState
+	cs.RUnlock()
+
+	if len(ps) <= 0 {
+		return 0, errNoPeers
+	}
+
+	// improve performance returning first highestBlock
+	if len(ps) == 1 {
+		for _, p := range ps {
+			if p.number == nil {
+				return 0, errNilBlockData
+			}
+			return p.number.Int64(), nil
+		}
```

Comment:
I don't think the for statement is needed here since we ensure the `len(ps) == 1`
```suggestion
               p := ps[0]
               if p.number == nil {
			return 0, errNilBlockData
		}
		return p.number.Int64(), nil
```

Reply:
Good approach. I will apply it after my shift 😄 thanks!

Reply:
I've changed this logic to get a better performance. I will mark you in the comment to review the new approach.